### PR TITLE
[PM-30264] - fix exact match dialog show logic

### LINKED ISSUE #18133: Forced Autofill and Fill & Save options are no longer longer available when match detection is set to "Exact"
### Steps To Reproduce

1. In the browser extension, set default URI match detection to "Exact".
2. In the browser, open a website that contains a login form.
3. In the extension, search for a login item that does not already contain a matching URI (and for which any existing URI fields have the corresponding match detection option set to "Default").
4. Click the "Options" menu (three vertical dots), and select **Autofill**.
5. In the browser extension, set default URI match detection to "Base domain".
6. Edit the login item from **Step 3**, clicking the :gear: icon for each saved "Website (URI)" field and changing the match detection to "Exact".
7. Click <kbd>Save</kbd>, and then click <kbd>\<</kbd> (Back) to return to the search results.
8. For the modified login item (from **Step 6**), click the "Options" menu (three vertical dots), and select **Autofill**.

### Expected Result

In **Step 4** and in **Step 8**, expect to be presented with the "Confirm Autofill" modal pop-up,  with options to click <kbd>Autofill and add this website</kbd> (saving the current browser URL as a new URI in the login item) or <kbd>Autofill without adding</kbd> (forcing the autofill on an non-matching website).

### Actual Result

There are no options to "Autofill and add this website" or to "Autofill without adding".  Instead, there is an error message that says "Cannot autofill: Default matching is set to 'Exact Match'. The current website does not exactly match the saved login details for this item."

### Screenshots or Videos

Image

### Additional Context

There is no good reason why selecting the forced autofill option from the Options menu should in any way be affected by URI match detection settings.

When this feature was being developed (see PR #17142), I had a brief [discussion](https://github.com/bitwarden/clients/pull/17142#issuecomment-3470730022) with @jaasen-livefront to explain this (and at that time, I was under the impression that my concerns had been understood).

The forced autofill options (previously called "Autofill" and "Autofill and Save" or "Fill and Save") were _never_ intended to be equivalent to the autofill functions that are available for the set of **autofill suggestions** (which is generated by applying the URI match detection rules), which can be readily autofilled using multiple existing methods (inline autofill menus; <kbd>Fill</kbd> button in Vault view; keyboard shortcuts; context menu; automatic autofill on page load).   In contrast, the special options "Autofill" and "Autofill and Save" (originally available at the bottom of the _View Login_ view, then subsequently relegated to the Options menu and renamed first "Autofill" and "Fill and Save", then "Autofill without adding" and "Autofill and add this website", respectively) were _always_  supposed to be options for **forced autofill**, specifically designed for situations in which the active URI does _not_ have an existing match in the vault.

This is explained in the [original PR](https://github.com/bitwarden/clients/pull/1367) by @eliykat, as well as the original [Feature Request](https://community.bitwarden.com/t/autofill-after-search/2808/) topic that lead to the creation of this functionality.  A [comment](https://community.bitwarden.com/t/autofill-after-search/2808/99?u=grb) by @danielleflinn from June 2024 affirmed that the browser extension UI refresh was not going to break the existing functionality: _"If the item you want to fill, does not appear in the suggested section (based on its URI matching), you will still be able to access the auto-fill action from the 3 dot “more” menu."_

I cannot conceive of any reasonable rationale for honoring the URI match detection settings when a user is deliberately intending to perform a **forced** autofill of a login item that has no valid match to the current website.  This bug creates a seriously degraded user experience, capriciously denying us the option to save new URI strings to existing login items, and defeating the whole purpose of the forced "Autofill" and "Autofill and Save" options.

### Operating System

Windows

### Operating System Version

Windows 11 (24H2)

### Web Browser

Chrome

### Browser Version

_No response_

### Environment Versions

Version: 2025.12.0; SDK: 'main (0107af7)'; Server version: 2025.12.1.

### Issue Tracking Info

- [x] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-30264

Fixes #18133

## 📔 Objective

This PR fixes the logic for presenting the "exact match" dialog when a user attempts to autofill. Tests were added to and updated to be clearer and more explicit about the new rules.

## 📸 Screenshots

https://github.com/user-attachments/assets/c8c45d39-142c-472c-bf08-4fbb4508ceab

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts [hunk 1]
```diff
@@ -204,12 +204,28 @@ export class ItemMoreOptionsComponent {
...
 
     const uris = cipher.login?.uris ?? [];
-    const cipherHasAllExactMatchLoginUris =
-      uris.length > 0 && uris.every((u) => u.uri && u.match === UriMatchStrategy.Exact);
-
     const uriMatchStrategy = await firstValueFrom(this.uriMatchStrategy$);
 
-    if (cipherHasAllExactMatchLoginUris || uriMatchStrategy === UriMatchStrategy.Exact) {
+    const cipherHasAllExactMatchLoginUris =
+      uris.length > 0 &&
```
Comment (nick-livefront) [inline apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts]: I still find it frustrating that you have to check for length when using `.every` - it makes sense _but_ it is not intuitive.

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts [hunk 2]
```diff
@@ -204,12 +204,28 @@ export class ItemMoreOptionsComponent {
...
+        u.uri && u.match === undefined
+          ? uriMatchStrategy !== UriMatchStrategy.Exact
+          : u.match !== UriMatchStrategy.Exact,
+      );
+
+    if (
+      // don't show the dialog if there is at least one non-exact match uri
+      !cipherHasOneNonExactMatchLoginUri &&
+      (cipherHasAllExactMatchLoginUris || uriMatchStrategy === UriMatchStrategy.Exact)
+    ) {
```
Comment (nick-livefront) [inline apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts]: ❓ Are `cipherHasOneNonExactMatchLoginUri` and `cipherHasAllExactMatchLoginUris` inverses of each other? When `cipherHasOneNonExactMatchLoginUri` is true, `cipherHasAllExactMatchLoginUris` has to be false and vice versa? 

Could this shortened to the below? The tests pass but I may be missing an edge case

```ts
const hasOnlyExactMatchUris =
      uris.length === 0
        ? uriMatchStrategy === UriMatchStrategy.Exact
        : uris.every((u) => (u.match ?? uriMatchStrategy) === UriMatchStrategy.Exact);
        
if (hasOnlyExactMatchUris) {
  ...
```

Comment (jaasen-livefront) [reply apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.ts]: @nick-livefront You're absolutely right. Brilliant suggestion! The tests are pretty exhaustive, I can't think of a case we're not covering.

**Review by nick-livefront — APPROVED** [review]

**Comment by bwbug** [comment]
@jaasen-livefront This PR seems to just change your internal unit tests, , but it definitely does not fix the issue that I reported in #18133.

Could you please re-open #18133, and could someone at Bitwarden please engage in some dialog to explain your thinking behind taking away forced autofill functionality from users who prefer the "Exact" URI match detection?   It is simply not a logical way for the extension to behave &mdash; if anything, the "Autofill and add this website" function is _more_ critical for users who have set URI match detection to "Exact" than for users who do not (because "Autofill and add this website" is the most convenient and least error-prone way to make sure that the saved URI does in fact _exactly_ match the URL of the current website)!
